cardtel/utils/playing.py

```diff
@@ -1,25 +1,73 @@
+import deuces
+import gameplay
+from django.db import transaction
+
+def play_cards(player, cards):
+    '''|player| attempts to play |cards| on the table.'''
+    if check_valid_play(player, cards):
+        for card in cards:
+            play_card(player, card)
+    else:
+        raise IllegalPlay
+
+@transaction.atomic
 def play_card(player, card):
-    raise NotImplementedError
+    '''|player| plays |card| on the table.'''
+    player.cards.remove(card)
+    player.game.table.add(card)
 
 def fold(player):
-    raise NotImplementedError
+    '''|player| folds, giving up his ability to play again this point.'''
+    player.has_folded = True
+    player.save()
 
-def check_valid_play(player, card):
-    raise NotImplementedError
+def check_valid_play(player, new_cards):
+    '''Check if |new_cards| makes the table better during |player|'s turn.'''
+    table_deuces_cards = get_deuces_cards(player.game.table.cards.all())
+    new_deuces_cards = get_deuces_cards(new_cards)
+    all_cards = new_deuces_cards + table_deuces_cards
+    current_score = player.game.table.best_hand_score
```

Comment:
every `.` call like in `player.game.table` is a DB hit to get the FK. Generally best practice to do `table = player.game.table` and then refer to table directly if it's gonna happen more than once